### ISSUE TITLE
diagram includes are not supported

### ISSUE THREAD

**Issue by ali-habibzadeh** [Enteee/plantuml-parser#25]
**Describe the bug**

In plantUML you can use `!include` and `!includeurl` for importing dependencies for a diagram (extensions, icons etc.) http://plantuml.com/preprocessing.

But if such imports are used the parser breaks.

```
return new peg$SyntaxError(
           ^
source-map-support.js:445
SyntaxError: Expected "'", "(", "/'", "@enduml", "[", "abstract ", "class ", "cloud", "component ", "database", "digraph", "enum ", "folder", "frame", "interface ", "namespace", "node", "note ", "package", "rectangle", "skinparam ", "state", "together ", "usecase ", [ \t], or [A-Za-z0-9._] but "}" found.
```

**To Reproduce**
1. Use parse with content or use parseFile with given uml
2. See error stack

**PlantUML**
```
@startuml
!includeurl https://raw.githubusercontent.com/RicardoNiepel/C4-PlantUML/release/1-0/C4.puml
!includeurl https://raw.githubusercontent.com/RicardoNiepel/C4-PlantUML/release/1-0/C4_Container.puml
!includeurl https://raw.githubusercontent.com/RicardoNiepel/C4-PlantUML/release/1-0/C4_Context.puml
!includeurl https://raw.githubusercontent.com/RicardoNiepel/C4-PlantUML/release/1-0/C4_Component.puml

System_Boundary(MetricsExecution, "Metrics Execution") {
    Component(apiRouter, "Express Routeer", "Api router")

    System_Ext(puppeteer, "puppeteer", "Headless Chrome Provider")

    Rel(apiRouter, PageRenderService, "url: string")
    Rel(PageRenderService, apiRouter, "IMetric<T>[]")
}
@enduml
```

**Expected behavior**
It should parse normally 

**Screenshots**
N/A

**Desktop (please complete the following information):**
 - OS: Mac OS Mojave
 - Browser Node
 - Version 12

**Additional context**
N/A

**Comment by Enteee** [comment]
Thank you for reporting this @ali-habibzadeh. 

`!includeurl` and some of its friends like `!log`, `!function` are all [PlantUML pre-preocessor featuers](http://plantuml.com/preprocessing). I fear that implementing them means opening pandora's box. Anyways I do understand that the parser should not raise an exception when it encounters not preprocessed PlantUML diagrams and still try to do its best with the diagram given.

Therefore I'd suggest:

0. I will add test fixtures for all examples in: [PlantUML pre-preocessor](http://plantuml.com/preprocessing)
1. The parser should ignore all lines starting with a `!` which will fix this bug.
2. In order to generate diagrams properly we still have to implement a PlantUML pre-processor (`preprocess()`), which takes a raw PlantUML diagram and returns a preprocessed version of that diagram. I will open a enhancement for this.

**Comment by Enteee** [comment]
Ok the simple test fixtures I pushed are parse in a low-effort manner. Which mean the `!includeurl` is just ignored by the parser. So what I wanted to do from my previous post:

> - [ ] The parser should ignore all lines starting with a ! which will fix this bug.

Does already work for simple one line preprocessor instructions.

The real reason for the parser exception is that without reading the includes this part:
```
System_Boundary(MetricsExecution, "Metrics Execution") {
    Component(apiRouter, "Express Routeer", "Api router")

    System_Ext(puppeteer, "puppeteer", "Headless Chrome Provider")

    Rel(apiRouter, PageRenderService, "url: string")
    Rel(PageRenderService, apiRouter, "IMetric<T>[]")
}
```
is completely undefined syntax.

This means in order to fix this bug we actually have to implement #26 to its full extend.

**Comment by Enteee** [comment]
@ali-habibzadeh since this bug requires effectively #26 in big parts, are you OK with the following plan?

1. I push your example to the test fixtures for #26 
2. I'll merge the simple fixtures I have created here (which don't show reproduce your bug) to master
3. We close this bug in favor for #26

**Comment by Enteee** [comment]
Closed, tracking in #26
